### PR TITLE
[Magiclysm] Dirty patches

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Mods "Added dozen of earthshaper's spells. Make earthshaping great again!"
#### Describe the solution
Added 9 earthshaper's spells:
Tier 0:
Reactive Armour - Explode your own stone skin, to deal massive damage to monster's nearby. (definitely not copypasted from divinity 2, trust me)
Petrifying Touch - okay damage, and good DoT and freeze on a melee distance
Shape a Boulder - spawn a boulder. you can use it as a table, as a workbench, as an 1 lvl anvil, or just decorate your home with it, its pretty nice (at least in ultica)
Tier 1
Shape A Tool - allow you to create a tools out of mana. the process is slow (and maybe i need to add material like rocks), but the result is really good
Tier 2
Impalement - good damage AoE with monster's slowdown
Weighting - slow down your attack speed for a bit, gives a pretty huge increase the melee damage
Crystal Wrapping - reflect some damage to monsters that attack you, gives additional cut damage in melee
Tier 3
Shape of Dust - de facto invisibility, but also create a cool sand effect around
Summon Earthshaper's Golem - spawn pretty dense golem, that is really good in cleaning low level areas

In addition:
Added some books, that contain some of the new spells
Edited a stone skill item to weight 100g instead of 20 kg: no more -50 speed after wearing the armor
Improved all "melee" spells - they require less time (you train it), and can be casted, when you have something in your hand (i tried to emulate one hand action, by using `NO_HANDS` and `SOMATIC` flags simultaneously, but no_hands overwrite everything related to arm encumbrance, so i decided to left only no_hands - range 1 is already a huge limitation for the spell, no need to make it worse)
Removed the requirement of ice cream for freezing touch spell
Optimize a freezing touch spell for a bit
Slightly edited the artificer's toolkit spell, to be more in line with another spells
#### Testing
The game works fine, each spell works as intended, loot is spawned
#### Additional context
I hope it will be reviewed asap, because i dont want to make a lot of branch conflicts with my next pr's (same as my previous *checks my list* 3 magiclysm PR)

### CODE REVIEW

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 1]
```diff
@@ -397,6 +397,114 @@
...
+    "name": { "str": "Scroll of Impalement", "str_pl": "Scrolls of Impalement" },
+    "description": "Summon a stone spikes underground, that penetrate your enemies, and nail them for a while.",
+    "use_action": { "type": "learn_spell", "spells": [ "eshaper_impalement" ] }
+  },
+  {
+    "type": "BOOK",
+    "copy-from": "spell_scroll",
+    "id": "spell_scroll_eshaper_spawn_tools",
+    "//": "Earthshaper spell",
+    "name": { "str": "Scroll of Shaping A Tool", "str_pl": "Scrolls of Shaping A Tool" },
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/items/spell_scrolls.json:459]:
```suggestion
    "name": { "str": "Scroll of Utility Molding", "str_pl": "Scrolls of Utility Molding" },
```

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 2]
```diff
@@ -397,6 +397,114 @@
...
+    "description": "Shape the basic working tools, using your own magic abilities and earth nearby.  Activate to change the shape of a tool.",
+    "use_action": { "type": "learn_spell", "spells": [ "eshaper_spawn_tools" ] }
+  },
+  {
+    "type": "BOOK",
+    "copy-from": "spell_scroll",
+    "id": "spell_scroll_eshaper_melee_damage",
+    "//": "Earthshaper spell",
+    "name": { "str": "Scroll of Weighting", "str_pl": "Scrolls of Weighting" },
+    "description": "Fill yourself with power of Earth, and make each your attack devastating as quake.",
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/items/spell_scrolls.json:469]:
```suggestion
    "description": "Fill yourself with power of Earth, and make each your attack as devastating as quake.",
```

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 3]
```diff
@@ -397,6 +397,114 @@
...
+    "name": { "str": "Scroll of Earthshaper's Golem", "str_pl": "Scrolls of Earthshaper's Golem" },
+    "description": "Shape a strong golem out of the stone.  It may help for a lot, but you have not enough energy to support it forever.",
+    "use_action": { "type": "learn_spell", "spells": [ "summon_eshaper_golem" ] }
+  },
+  {
+    "type": "BOOK",
+    "copy-from": "spell_scroll",
+    "id": "spell_scroll_earthshaper_boulder",
+    "//": "Earthshaper spell",
+    "name": { "str": "Scroll of a Boulder", "str_pl": "Scrolls of a Boulder" },
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/items/spell_scrolls.json:504]:
```suggestion
    "name": { "str": "Scroll of Summon Boulder", "str_pl": "Scrolls of Summon Boulder" },
```

File: data/mods/Magiclysm/items/spellbooks.json [hunk 4]
```diff
@@ -311,9 +311,42 @@
...
+      "spells": [ "pertifying_touch", "earthshaper_stoneskin", "reactive_armor", "eshaper_melee_damage", "eshaper_crystal_wrap" ]
+    }
+  },
+  {
+    "id": "eshaper_spellbook_tool",
+    "type": "BOOK",
+    "category": "manuals",
+    "name": { "str": "One Spell is All You Need!", "str_pl": "copies of One Spell is All You Need!" },
+    "//": "1 earthshaper spell. yes, the whole book is only for one, that's the idea",
+    "description": "The book ( more like advertising booklet ) promote one spell, that allow you to create a different tools for different purposes.  Surprisingly, it works as the book says - you can change the shape of your tool, depending on your needs.",
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/items/spellbooks.json:341]:
```suggestion
    "description": "This advertising booklet promotes one spell that allows you to create a different tool for different purposes.  Surprisingly, it works as the book says - you can change the shape of your tool, depending on your needs.",
```

File: data/mods/Magiclysm/monsters/golems.json [hunk 5]
```diff
@@ -122,5 +119,35 @@
     "special_attacks": [ { "type": "spell", "spell_data": { "id": "gas_attack", "min_level": 5 }, "cooldown": 60 } ],
     "death_function": { "corpse_type": "BROKEN" },
     "flags": [ "SEES", "NO_BREATHE", "ACIDPROOF", "LOUDMOVES" ]
+  },
+  {
+    "id": "mon_eshaper_golem",
+    "type": "MONSTER",
+    "name": "earthshaper's golem",
+    "description": "This golem is easy to separate from other golems - the shiny earthshaper's emblem shine on his smooth body, and his eyes and hands are made out of purple crystals.",
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/monsters/golems.json:127]:
```suggestion
    "description": "This golem is easy to separate from other golems - the metallic earthshaper's emblem shines on its smooth body, and it's eyes and hands are made out of amethyst.",
```
I don't think golems have a gender.

File: data/mods/Magiclysm/ter_fur_transform/ter_fur_transform.json [hunk 6]
```diff
@@ -102,6 +102,17 @@
...
     "furniture": [ { "result": "f_null", "valid_flags": [ "TRANSPARENT", "WALL" ] } ]
   },
+  {
+    "type": "ter_furn_transform",
+    "id": "earthshaper_boulder",
+    "furniture": [
+      {
+        "result": "f_boulder_earthshaper",
+        "valid_furniture": [ "f_null" ],
+        "message": "The ground shakes, when a section of unnaturally smooth stone rising up."
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/ter_fur_transform/ter_fur_transform.json:112]:
```suggestion
        "message": "The ground shakes, and a section of unnaturally smooth stone rises to the surface."
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 7]
```diff
@@ -356,5 +356,286 @@
...
+    "max_aoe": 1,
+    "flags": [ "SOMATIC", "VERBAL", "CONCENTRATE" ],
+    "min_duration": 12000,
+    "max_duration": 360000,
+    "duration_increment": 12000
+  },
+  {
+    "id": "earthshaper_boulder",
+    "type": "SPELL",
+    "name": "Shape a Boulder",
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/Spells/earthshaper.json:623]:
```suggestion
    "name": "Summon Boulder",
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 8]
```diff
@@ -356,5 +356,286 @@
...
+    "min_aoe": 1,
+    "max_aoe": 7,
+    "aoe_increment": 0.3,
+    "min_damage": -1000,
+    "max_damage": -1
+  },
+  {
+    "id": "eshaper_spawn_tools",
+    "type": "SPELL",
+    "name": "Shape A Tool",
```
Comment (MNG-cataclysm) [inline data/mods/Magiclysm/Spells/earthshaper.json:483]:
```suggestion
    "name": "Utility Molding",
```

**Review by MNG-cataclysm — CHANGES_REQUESTED** [review]

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 9]
```diff
@@ -356,5 +356,286 @@
...
+    "max_level": 25,
+    "min_duration": 14400,
+    "max_duration": 360000,
+    "duration_increment": 14400
+  },
+  {
+    "id": "eshaper_crystal_wrap",
+    "type": "SPELL",
+    "name": "Crystal Wrapping",
+    "description": "Your cover your body with small layer of razor sharp crystals.  It makes monster's, that attack you, take some damage in return, and gives you some additional cut damage in melee.",
```
[the remark this answers is not in the document]
Comment (RAldrich) [reply data/mods/Magiclysm/Spells/earthshaper.json:534]:
```suggestion
    "description": "Your cover your body with small layer of razor sharp crystals.  It makes monsters that attack you take some damage in return, and gives you some additional cut damage in melee.",
```

File: data/mods/Magiclysm/effects/effects.json [hunk 10]
```diff
@@ -446,6 +446,35 @@
...
+    "rating": "good",
+    "show_intensity": false,
+    "enchantments": [ { "values": [ { "value": "ATTACK_SPEED", "multiply": -0.25 }, { "value": "MELEE_DAMAGE", "multiply": 1.2 } ] } ]
+  },
+  {
+    "id": "eshaper_crystal_wrap",
+    "type": "effect_type",
+    "name": [ "Crystal Wrapping" ],
+    "desc": [ "Thin layer of almost living crystals, that rapidly grows and moves on the body." ],
+    "apply_message": "You feel a light, purple material cover your body and clothes.",
```
[the remark this answers is not in the document]
Comment (GuardianDll) [reply data/mods/Magiclysm/effects/effects.json:467]: I don't want to connect it to some existed mineral tbh, prefer to left it abstract. Maybe will make some new material only for earthshaper, who knows

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 11]
```diff
@@ -356,5 +356,286 @@
...
+  {
+    "id": "pertifying_touch_moves",
+    "type": "SPELL",
+    "name": "Petrifying Touch Moves Removing",
+    "description": "Delete some amount of moves from monster, and inflict DoT.  You can see it only in debug mode.",
+    "valid_targets": [ "hostile" ],
+    "flags": [ "SILENT", "NO_EXPLOSION_SFX" ],
+    "effect": "mod_moves",
+    "shape": "blast",
+    "damage_type": "stab",
```
Comment (GuardianDll) [inline data/mods/Magiclysm/Spells/earthshaper.json:396]:
```suggestion
```

File: data/mods/Magiclysm/Spells/earthshaper.json [hunk 12]
```diff
@@ -356,5 +356,286 @@
...
+    "damage_type": "cut",
+    "min_range": 1,
+    "max_range": 1
+  },
+  {
+    "id": "shape_of_dust",
+    "type": "SPELL",
+    "name": "Shape of Dust",
+    "description": "Hide yourself between the sand gust.  It's harder to hit you, and monster's can't spot you.",
+    "//": "Fuck its broken. I thought you can make monster hard to see you, so you will be able to hide from them beyond the sand, but of course invis make monsters even more dumb, and they start to stoically ignore your existence.",
```
[the remark this answers is not in the document]
Comment (GuardianDll) [reply data/mods/Magiclysm/Spells/earthshaper.json:573]: It adds, but in this case its just a weak defensive item, that do not really make you "harder to spot", so i added an invisibility effect. Guess, if there would be a way to impale monster's vision, ill replace the invis with this effect, but now we have what we have

**Pull request merged**